src/SumoCoders/FrameworkCoreBundle/Form/Extension/ButtonTypeIconExtension.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Form\Extension;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+class ButtonTypeIconExtension extends AbstractTypeExtension
+{
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->setAttribute('icon', $options['icon']);
+    }
+
+    /**
+     * @param FormView      $view
+     * @param FormInterface $form
+     * @param array         $options
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['icon'] = $options['icon'];
+    }
+
+    /**
+     * @param OptionsResolverInterface $resolver
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(array(
+            'icon' => null,
+        ));
+    }
+
+    /**
+     * Returns the name of the type being extended.
+     *
+     * @return string The name of the type being extended
+     */
+    public function getExtendedType()
+    {
+        return ButtonType::class; // Extend the button field type
```

Comment:
we zetten trouwens ook nooit onze comments achter onze lijn, maar er boven. (maar ik zou het hier eigenlijk in de docblock van de method zetten)